diff --git a/node_modules/x-data-spreadsheet/src/index.d.ts b/node_modules/x-data-spreadsheet/src/index.d.ts
index 948a8e9..3425cb0 100644
--- a/node_modules/x-data-spreadsheet/src/index.d.ts
+++ b/node_modules/x-data-spreadsheet/src/index.d.ts
@@ -182,6 +182,12 @@ declare module 'x-data-spreadsheet' {
      * @param callback
      */
     change(callback: (json: Record<string, any>) => void): this;
+
+    /**
+     * bind handler to change event, including data change and user actions
+     * @param callback
+     */
+    onAddSheet(callback: () => void): this;
     /**
      * set locale
      * @param lang
diff --git a/node_modules/x-data-spreadsheet/src/index.js b/node_modules/x-data-spreadsheet/src/index.js
index 479a0b7..980db41 100644
--- a/node_modules/x-data-spreadsheet/src/index.js
+++ b/node_modules/x-data-spreadsheet/src/index.js
@@ -20,6 +20,7 @@ class Spreadsheet {
     this.bottombar = this.options.showBottomBar ? new Bottombar(() => {
       const d = this.addSheet();
       this.sheet.resetData(d);
+      this.sheet.trigger('add-sheet', "")
     }, (index) => {
       const d = this.datas[index];
       this.sheet.resetData(d);
@@ -120,6 +121,11 @@ class Spreadsheet {
     return this;
   }
 
+  onAddSheet(cb) {
+    this.sheet.on('add-sheet', cb);
+    return this;
+  }
+
   static locale(lang, message) {
     locale(lang, message);
   }
