diff --git a/node_modules/x-data-spreadsheet/src/component/event.js b/node_modules/x-data-spreadsheet/src/component/event.js
index cdbc0fa..cb2a9db 100644
--- a/node_modules/x-data-spreadsheet/src/component/event.js
+++ b/node_modules/x-data-spreadsheet/src/component/event.js
@@ -74,6 +74,7 @@ export function bindTouch(target, { move, end }) {
       starty = pageY;
     }
     evt.preventDefault();
+    evt.stopPropagation();
   });
   bind(target, 'touchend', (evt) => {
     if (!end) return;
diff --git a/node_modules/x-data-spreadsheet/src/component/sheet.js b/node_modules/x-data-spreadsheet/src/component/sheet.js
index 086b8c4..cd01f6e 100644
--- a/node_modules/x-data-spreadsheet/src/component/sheet.js
+++ b/node_modules/x-data-spreadsheet/src/component/sheet.js
@@ -77,11 +77,11 @@ function selectorSet(multiple, ri, ci, indexesUpdated = true, moving = false) {
   const cell = data.getCell(ri, ci);
   if (multiple) {
     selector.setEnd(ri, ci, moving);
-    this.trigger('cells-selected', cell, selector.range);
+    this.trigger('cells-selected', this.data, selector.range);
   } else {
     // trigger click event
     selector.set(ri, ci, indexesUpdated);
-    this.trigger('cell-selected', cell, ri, ci);
+    this.trigger('cell-selected', this.data, ri, ci);
   }
   toolbar.reset();
   table.render();
diff --git a/node_modules/x-data-spreadsheet/src/index.d.ts b/node_modules/x-data-spreadsheet/src/index.d.ts
index 948a8e9..7fe36b3 100644
--- a/node_modules/x-data-spreadsheet/src/index.d.ts
+++ b/node_modules/x-data-spreadsheet/src/index.d.ts
@@ -45,18 +45,18 @@ declare module 'x-data-spreadsheet' {
   export interface SpreadsheetEventHandler {
     (
       envt: CELL_SELECTED,
-      callback: (cell: Cell, rowIndex: number, colIndex: number) => void
+      callback: (sheetData: Record<string, any>, rowIndex: number, colIndex: number) => void
     ): void;
     (
       envt: CELLS_SELECTED,
       callback: (
-        cell: Cell,
+        sheetData: Record<string, any>,
         parameters: { sri: number; sci: number; eri: number; eci: number }
       ) => void
     ): void;
     (
       evnt: CELL_EDITED,
-      callback: (text: string, rowIndex: number, colIndex: number) => void
+      callback: (sheetData: Record<string, any>, rowIndex: number, colIndex: number) => void
     ): void;
   }
 
@@ -182,6 +182,19 @@ declare module 'x-data-spreadsheet' {
      * @param callback
      */
     change(callback: (json: Record<string, any>) => void): this;
+
+    /**
+     * 添加sheet监听回调
+     * @param callback
+     */
+    onAddSheet(callback: () => void): this;
+
+     /**
+     * 重命名Sheet
+     * @param callback
+     */
+     onRenameSheet(callback: () => void): this;
+
     /**
      * set locale
      * @param lang
diff --git a/node_modules/x-data-spreadsheet/src/index.js b/node_modules/x-data-spreadsheet/src/index.js
index 479a0b7..96d0f3a 100644
--- a/node_modules/x-data-spreadsheet/src/index.js
+++ b/node_modules/x-data-spreadsheet/src/index.js
@@ -20,6 +20,7 @@ class Spreadsheet {
     this.bottombar = this.options.showBottomBar ? new Bottombar(() => {
       const d = this.addSheet();
       this.sheet.resetData(d);
+      this.sheet.trigger('add-sheet', "")
     }, (index) => {
       const d = this.datas[index];
       this.sheet.resetData(d);
@@ -27,6 +28,7 @@ class Spreadsheet {
       this.deleteSheet();
     }, (index, value) => {
       this.datas[index].name = value;
+      this.sheet.trigger('rename-sheet', value)
     }) : null;
     this.data = this.addSheet();
     const rootEl = h('div', `${cssPrefix}`)
@@ -120,6 +122,16 @@ class Spreadsheet {
     return this;
   }
 
+  onAddSheet(cb) {
+    this.sheet.on('add-sheet', cb);
+    return this;
+  }
+
+  onRenameSheet(cb) {
+    this.sheet.on('rename-sheet', cb);
+    return this;
+  }
+
   static locale(lang, message) {
     locale(lang, message);
   }
